Please review, failing test:
```
________________________________________________________________________________

 sympy/tensor/tests/test_tensor_operators.py:test_expand_partial_derivative_constant_factor_rule 

Traceback (most recent call last):

  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/sympy-1.6.dev0-py3.6.egg/sympy/testing/runtests.py", line 1328, in _timeout

    function()

  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/sympy-1.6.dev0-py3.6.egg/sympy/tensor/tests/test_tensor_operators.py", line 203, in test_expand_partial_derivative_constant_factor_rule

    nneg*PartialDerivative(A(i), D(j))

AssertionError

 tests finished: 1326 passed, 1 failed, 77 skipped, 185 expected to fail, 

3 expected to fail but passed, in 1258.65 seconds 

DO *NOT* COMMIT!

Traceback (most recent call last):

  File "<stdin>", line 4, in <module>

Exception: Tests failed
```

--- a/sympy/stats/frv_types.py
+++ b/sympy/stats/frv_types.py
@@ -90,9 +90,17 @@ def FiniteRV(name, density):
     return rv(name, FiniteDistributionHandmade, density)
 
 class DiscreteUniformDistribution(SingleFiniteDistribution):
+
+    @staticmethod
+    def check(*args):
+        _value_check(len(set(args)) == len(args), """Repeated args detected
+        but set expected. If you want a distribution that has different
+        weightings for each item consider using FiniteRV(%s, %s)."""
+        % ('Symbol("rv")', set(args)))
+
     @property
     def p(self):
-        return Rational(1, len(set(self.args)))
+        return Rational(1, len(self.args))
 
     @property  # type: ignore
     @cacheit

--- a/sympy/stats/frv_types.py
+++ b/sympy/stats/frv_types.py
@@ -104,7 +104,7 @@ class DiscreteUniformDistribution(SingleFiniteDistribution):
         return set(self.args)
 
     def pmf(self, x):
-        if x in set(self.args):
+        if x in self.args:
             return self.p
         else:
             return S.Zero

--- a/sympy/stats/frv_types.py
+++ b/sympy/stats/frv_types.py
@@ -93,10 +93,15 @@ class DiscreteUniformDistribution(SingleFiniteDistribution):
 
     @staticmethod
     def check(*args):
-        _value_check(len(set(args)) == len(args), """Repeated args detected
-        but set expected. If you want a distribution that has different
-        weightings for each item consider using FiniteRV(%s, %s)."""
-        % ('Symbol("rv")', set(args)))
+        from sympy.utilities.iterables import multiset
+        from sympy.utilities.misc import filldedent
+        weights = multiset(args)
+        _value_check(len(set(args)) == len(args),
+            filldedent("""
+            Repeated args detected but set expected. For a
+            distribution having different weights for each
+            item use the following:\nFiniteRV(%s, %s)."""
+            % ('X', weights)))
 
     @property
     def p(self):

--- a/sympy/stats/tests/test_finite_rv.py
+++ b/sympy/stats/tests/test_finite_rv.py
@@ -46,12 +46,7 @@ def test_discreteuniform():
     assert characteristic_function(X)(t) == exp(I*a*t)/3 + exp(I*b*t)/3 + exp(I*c*t)/3
     assert moment_generating_function(X)(t) == exp(a*t)/3 + exp(b*t)/3 + exp(c*t)/3
     # issue 18611
-    Z = DiscreteUniform('Z', [a, a, a, b, b, c])
-    assert dict(density(Z).items()) == {a: S('1/3'), b: S('1/3'), c: S('1/3')}
-    assert E(Z) == (a + b + c)/3
-    Z = DiscreteUniform('Z', [1, 1, 1, 2, 2, 3, 3, 3, 4])
-    assert dict(density(Z).items()) == {1: S('1/4'), 2: S('1/4'), 3: S('1/4'), 4: S('1/4')}
-    assert cdf(Z) == {1: S('1/4'), 2: S('1/2'), 3: S('3/4'), 4: 1}
+    raises(ValueError, lambda: DiscreteUniform('Z', [a, a, a, b, b, c]))
 
 def test_dice():
     # TODO: Make iid method!

